src/pages/Register/Register.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import axios from "axios";
+import "./Register.css";
+import { useAuth } from "../../context/auth-context";
+import { Link, useNavigate } from "react-router-dom";
+import RegisterPic from "../../assets/register.svg";
+import Navbar from "../../Components/Navbar/Navbar";
+
+const Register = () => {
+  const { setUser, setToken } = useAuth();
+  const navigate = useNavigate();
+  const registerHandler = async () => {
+    try {
+      const res = await axios.post("/api/auth/signup", {
+        firstName,
+        lastName,
+        email,
+        password,
+      });
+      console.log(res);
+      if (res.status === 201 && res.data.encodedToken) {
+        setToken(res.data);
+        setUser(res.data.foundUser);
+        navigate("/mynotes");
+      }
+    } catch (error) {
+      console.log(error);
+      console.log(error.response);
+    }
+  };
+
+  const [firstName, setFirstName] = useState("John");
+  const [lastName, setLastName] = useState("Doe");
+  const [email, setEmail] = useState("john@gmail.com");
+  const [password, setPassword] = useState("john123");
+
```

Comment:
Here instead of creating separate useState for each. You can can do it with just one by doing something like this:
```
const [userDetails,setUserDetails] = useState({
firstName:"John",
lastName:"Doe",
email:"john@gmail,.com",
password:"john123"
})
```

Reply:
Great suggestion, will do this in refactoring post submission!